src/Cmd/Build.cc

```diff
@@ -33,29 +33,45 @@ const Subcmd BUILD_CMD =
         .setMainFn(buildMain);
 
 int
-buildImpl(std::string& outDir, const bool isDebug) {
-  const auto start = std::chrono::steady_clock::now();
-
-  const BuildConfig config = emitMakefile(isDebug, /*includeDevDeps=*/false);
-  outDir = config.outBasePath;
-
-  const std::string& packageName = getPackageName();
+runBuildCommand(
+    const std::string& outDir, const BuildConfig& config,
+    const std::string& targetName
+) {
   const Command makeCmd = getMakeCommand().addArg("-C").addArg(outDir).addArg(
-      (config.outBasePath / packageName).string()
+      (config.outBasePath / targetName).string()
   );
   Command checkUpToDateCmd = makeCmd;
   checkUpToDateCmd.addArg("--question");
 
   int exitCode = execCmd(checkUpToDateCmd);
   if (exitCode != EXIT_SUCCESS) {
-    // If packageName binary is not up-to-date, compile it.
+    // If targetName binary is not up-to-date, compile it.
     logger::info(
-        "Compiling", "{} v{} ({})", packageName, getPackageVersion().toString(),
+        "Compiling", "{} v{} ({})", targetName, getPackageVersion().toString(),
         getProjectBasePath().string()
-
     );
     exitCode = execCmd(makeCmd);
   }
+  return exitCode;
+}
+
+int
+buildImpl(std::string& outDir, const bool isDebug) {
+  const auto start = std::chrono::steady_clock::now();
+
+  const BuildConfig config = emitMakefile(isDebug, /*includeDevDeps=*/false);
+  outDir = config.outBasePath;
+
+  const std::string& packageName = getPackageName();
+  int exitCode = 0;
+  if (config.isExecutable()) {
+    exitCode = runBuildCommand(outDir, config, packageName);
+  }
+
+  if (config.isLibrary() && exitCode == 0) {
+    const std::string libName = getLibraryName();
```

Comment:
I think you can add a `getLibName` method in `BuildConfig`, and then:
```suggestion
    const std::string libName = config.getLibName();
```